File: Bit Manipulation/Easy/461. Hamming Distance.c
/**
 * Problem: Compute Hamming Distance
 * Description: Given two integers x and y, calculate the Hamming distance between them. 
 * The Hamming distance is the number of positions at which the corresponding bits are different.
 * 
 * @param x: The first integer.
 * @param y: The second integer.
 * @return: The Hamming distance between x and y.
 */
int hammingDistance(int x, int y) {
    int c = x ^ y; // XOR operation gives the bits that are different between x and y
    int count = 0;

    // Count the number of set bits (1s) in the result of XOR operation
    while (c > 0) {
        if (c % 2 == 1) {
            count++;
        }
        c = c / 2; // Move to the next bit
    }

    return count;
}

## Time Complexity: O(1), as the number of bits in the integers is constant.
## Space Complexity: O(1), as we use only a constant amount of extra space.

File: Bit Manipulation/Easy/461. Hamming Distance.py
# Problem: Calculate Hamming Distance
# Description: Given two integers x and y, return the Hamming distance between them, which is the number of positions 
# at which the corresponding bits are different.

class Solution(object):
    def hammingDistance(self, x, y):
        """
        Calculates the Hamming distance between two integers x and y. The Hamming distance is defined as the number of 
        positions at which the corresponding bits are different.

        :param x: Integer, the first integer.
        :param y: Integer, the second integer.
        :return: Integer, the Hamming distance between x and y.
        """
        return bin(x ^ y).replace('0b', "").count('1')

## Time Complexity: O(1), as the bitwise XOR operation and counting bits are constant-time operations.
## Space Complexity: O(1), as we only use a constant amount of extra space.

# Example usage:
# solution = Solution()
# print(solution.hammingDistance(1, 4))  # Expected to return 2
